docs/RuntimeAttributes.md

```diff
@@ -27,6 +25,37 @@ workflow jes_workflow {
 }
 ```
 
+
+## Recognized Runtime attributes and Backends
+
+Cromwell recognizes certain runtime attributes and has the ability to format these for some [Backends](/backends/Backends). See the table below for common attributes that apply to _most_ backends.
+
+
+> [Shared Filesystem backend](/backends/HPC#shared-filesystem) (SFS) is fully configurable and thus these attributes do not apply universally.
+
```

Comment:
Feel free to ignore this as it's not really in scope for your efforts but the `SGE` column on this table is a holdover from when the HPC/Config/whatever one wants to call it backend **was** the `SGE` backend.

Reply:
I renamed it SFS, but do you think I should just remove the column and add specific section for SFS? I briefly mention it within the ```### `memory` ``` section that you can include things like `memory_mb`. 

I think it might be useful to add the _exposed attribute_ instead of an `x`, and then add a bunch of these parameters the [HPC Backends](https://cromwell.readthedocs.io/en/stable/backends/HPC/) page to show how these attributes are exposed (including others like `${script}, ${stdout}, ...`), a general way to write a config (with attributes) and then link to the example configs. (Some of the SFS attributes are listed on the [SGE config page](https://cromwell.readthedocs.io/en/stable/backends/SGE/)).

Maybe something like this:

| Runtime Attribute    | LOCAL |  Google Cloud  | AWS Batch |  SFS  |
| -------------------- |:-----:|:-----:|:-----:|:------:|
| [cpu](#cpu)                  |       |   x   |   x   |  `cpu`  |
| [memory](#memory)                              |       |   x   |   x   |  `memory_mb` / `memory_gb`  |
| [disks](#disks)                                |       |   x   |       |  `disks`  |
| [docker](#docker)                              |   x   |   x   |   x   |  `docker` (see below)  |
| [maxRetries](#maxretries)                      |   x   |   x   |   x   | _N/A_ |
| [continueOnReturnCode](#continueonreturncode) |   x   |   x   |   x   |  _N/A_  |
| [failOnStderr](#failonstderr)                  |   x   |   x   |   x   |  _N/A_  |

The docker section already links to our [Containers](https://cromwell.readthedocs.io/en/stable/tutorials/Containers/) guide. I think it would be worth adding some more details in how containers are supported in cloud, and then more specifically the SFS docker attributes as there are a ton: `docker_cwd, docker_out, docker_err` etc.

Reply:
Hey @geoffjentry, just wondering what you think of this, and what changes you'd recommend for this PR.

Reply:
Sorry, I was away. On the surface I like that proposal, although I tend to prefer HPC over SFS just to separate the concept from local. 

The runtime attributes are configurable for the config/sfs backend. I'm assuming you picked out the most typical (ie what's in our examples). I think that's fine, just make sure that's spelled out

Reply:
What I'm going for in the table is these are the ones that Cromwell _recognises_ and processes. Other runtime attributes aren't processed like this.
- `cpu` has a default of 1 if not given
- `memory` is parsed and (at least) `memory_mb` and `memory_gb` are exposed.
- `disks` I'll comment out nothing actually happens here (that's what the interpunct is for, but I'll explain it better). 
- `docker` is looked up, and converted into a digest and used for call caching 
- The interpunct I'll turn into a star and explain it below, moving the HPC quote below the table.

Reply:
I've changed this in (8cc7be9984f9ffad102ba2ad62e91925ef0caeb0) to read:

> | Runtime Attribute    | LOCAL |  Google Cloud  | AWS Batch |  HPC  |
> | -------------------- |:-----:|:-----:|:-----:|:------:|
> | [cpu](#cpu)                  |       |   x   |   x   |  `cpu`  |
> | [memory](#memory)                              |       |   x   |   x   |  `memory_mb` / `memory_gb`  |
> | [disks](#disks)                                |       |   x   |       |  *  |
> | [docker](#docker)                              |   x   |   x   |   x   |  `docker` (see below)  |
> | [maxRetries](#maxretries)                      |   x   |   x   |   x   | * |
> | [continueOnReturnCode](#continueonreturncode) |   x   |   x   |   x   | * |
> | [failOnStderr](#failonstderr)                  |   x   |   x   |   x   |  *  |
> 
> 
> > `*` The HPC [Shared Filesystem backend](/backends/HPC#shared-filesystem) (SFS) is fully configurable and any number of attributes can be exposed. Cromwell recognizes some of these attributes (`cpu`, `memory` and `docker`) and parses them into the attribute listed in the table which can be used within the HPC backend configuration.

In the next couple of weeks I'll aim to submit a PR around:
- HPC support for Dockers (specifically listing the exposed attributes `docker_*`.